DonoGear/app/src/main/java/com/example/donogear/actionpages/HomePageFragment.java

```diff
@@ -27,6 +64,135 @@ public HomePageFragment() {
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home_page, container, false);
+        view = inflater.inflate(R.layout.fragment_home_page, container, false);
+        activity = (MainActivity) getActivity();
+
+        System.out.println("Home Page called");
+        setHasOptionsMenu(true);
+        handler = new Handler();
+
+        Runnable proceedsRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if(activity.hasAllImages && activity.hasAllData) {
+                    System.out.println("Initialize");
+                    initializeLayout();
+                }
+                else {
+                    handler.postDelayed(this, 100);
+                }
+            }
+        };
+        handler.post(proceedsRunnable);
+        return view;
+
+    }
+
+    /**
+     * Initializing layout and setting adapters of trending items, announcements and myInterests item
+     */
+    private void initializeLayout() {
```

Comment:
This method is pretty large. It might be worthwhile to break this down into one or more additional methods because this is harder to follow